File: ExamenMoria_Java/src/main/java/factories/salasFactory/SalasFactory.java
package factories.salasFactory;

import models.cola.ColaSala;
import models.salas.Sala;
import models.salas.SalaAccion;
import models.salas.SalaMagica;

import java.util.ArrayList;
import java.util.List;

import static utils.Utils.randomNumberGeneratorForIntegers;

public class SalasFactory {

    private static int contador = 1;

    public static ColaSala randomSalasGenerator(int poderMaligno, int flechas, int enemigos){
        List<Sala> salas = new ArrayList<>();

        for (int i=0; i<36; i++){
            int random = randomNumberGeneratorForIntegers(1, 99);

            if (random >= 1 && random <= 33){
                salas.add(
                        new SalaMagica(contador++, randomNumberGeneratorForIntegers(1, poderMaligno))
                );
            }else {
                if (random >= 34 && random <= 66){
                    salas.add(
                            new SalaAccion(contador++, randomNumberGeneratorForIntegers(1, flechas), randomNumberGeneratorForIntegers(1, enemigos))
                    );
                }else {
                    salas.add(
                            new Sala(contador++)
                    );
                }
            }
        }

        return new ColaSala(salas);
    }
}

File: ExamenMoria-Python/projecto/factories/salas/SalasFactory.py
import sys, os

path = os.path.abspath("projecto/models/salasPython")

sys.path.append(path)
import Sala, SalaMagica, SalaAccion
import random

def createSalas(poderMaligno, flechas, enemigos) -> list:
    salas = list()

    contador = 0

    while contador < 36:
        contador += 1

        rng = random.randint(1,99)

        if rng <= 33:
            salas.append(
                SalaMagica.SalaMagica(
                contador, random.randint(1, poderMaligno)
                )
            )
        elif rng <= 66:
            salas.append(
                SalaAccion.SalaAccion(
                contador, random.randint(1, flechas), random.randint(1, enemigos)
                )
            )
        else:
            salas.append(
                Sala.Sala(contador)
            )

    return salas